.github/workflows/deploy-hubs.yaml

```diff
@@ -6,121 +6,129 @@ on:
-          - cluster_name: uwhackweeks
-            provider: aws
-          - cluster_name: meom-ige
-            provider: gcp
-          - cluster_name: pangeo-hubs
-            provider: gcp
-          - cluster_name: leap
-            provider: gcp
-          - cluster_name: utoronto
-            provider: kubeconfig
-          - cluster_name: azure.carbonplan
-            provider: kubeconfig
+    outputs:
+      support-and-staging-matrix-jobs: ${{ steps.generated-jobs.outputs.support-and-staging-matrix-jobs }}
+      prob-hub-matrix-jobs: ${{ steps.generated-jobs.outputs.prod-hub-matrix-jobs }}
 
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
-
-      - name: Check if any of our base files have changed
-        uses: dorny/paths-filter@v2
-        id: base_files
         with:
-          filters: |
-            files:
-              - "deployer/**"
-              - "helm-charts/**"
-              - "requirements.txt"
-              - "config/secrets.yaml"
-              - ".github/workflows/deploy-hubs.yaml"
-              - ".github/actions/deploy/*"
-
-      - name: Check which cluster directory has changes (if any)
-        uses: dorny/paths-filter@v2
-        id: config_files
+          # Fetch-depth is required for the jitterbit/get-changed-files action
+          # https://github.com/jitterbit/get-changed-files/issues/24#issuecomment-1047442168
+          fetch-depth: 0
+
+      - name: Install Python 3.9
+        uses: actions/setup-python@v3
         with:
-          filters: |
-            hub_config:
-              - "config/clusters/${{ matrix.cluster_name }}/**"
-
-      - name: Setup gcloud
-        if: |
-          (steps.base_files.outputs.files == 'true') ||
-          (steps.config_files.outputs.hub_config == 'true')
-        uses: google-github-actions/setup-gcloud@v0
+          python-version: "3.9"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.txt
+
+      - name: Identify files that have been added or modified
+        uses: jitterbit/get-changed-files@v1
+        id: changed-files
         with:
-          version: "290.0.1"
-          # This is used for KMS only
-          project_id: two-eye-two-see
-          service_account_key: ${{ secrets.GCP_KMS_DECRYPTOR_KEY }}
-          export_default_credentials: true
-
-      - name: Setup helm
-        if: |
-          (steps.base_files.outputs.files == 'true') ||
-          (steps.config_files.outputs.hub_config == 'true')
-        uses: azure/setup-helm@v2.0
-
-      - name: Setup sops
-        if: |
-          (steps.base_files.outputs.files == 'true') ||
-          (steps.config_files.outputs.hub_config == 'true')
-        uses: mdgreenwald/mozilla-sops-action@v1
+          format: space-delimited
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate matrix jobs
+        if: github.event_name != 'pull_request'
```

Comment:
It is serving a purpose, we run a different last step in this job if triggered by a PR.

When triggered by a PR, we pretty-print the matrix jobs by running this step:

https://github.com/2i2c-org/infrastructure/pull/1159/files#diff-b04e1f867bb3bf0588d7d7e8f3337f0206eebc2337a56cc569d3146a4a2edc0fR65-R71

When triggered by a push to the default branch, the matrix jobs are sent to the next jobs via `::set-output`:

https://github.com/2i2c-org/infrastructure/pull/1159/files#diff-b04e1f867bb3bf0588d7d7e8f3337f0206eebc2337a56cc569d3146a4a2edc0fR61-R63

Reply:
Ah, the "pretty print matrix jobs" step below also has `if: github.event_name != 'pull_request'` rather than `if: github.event_name == 'pull_request'`.

I think it can be nice to have a pretty printed summary alongside doing or not doing `::set-output` no matter what btw. If easy to adjust, I suggest to make us do `::set-output` + `--pretty-print` at all times and then like we have it now - we ignore any outputs if we have a pull_request event later.

Reply:
The following commits should have addressed this:

- [f7dadad](https://github.com/2i2c-org/infrastructure/pull/1159/commits/f7dadadf3868b6ecb969f62026663edf99ed8b7b)
- [4028548](https://github.com/2i2c-org/infrastructure/pull/1159/commits/4028548b4c49205dbaa3ed8587cfa0e190cf6949)
- [c388e6e](https://github.com/2i2c-org/infrastructure/pull/1159/commits/c388e6e984f79fa1b9e9c4e3e0fc78a66dc0e3b0)